### DISCUSSION TITLE
Feature Suggestion: Smokemonster pack integration (ie: easy generation of Flash Cart Packs)

### DISCUSSION THREAD

**Discussion by joncamp** [retronas/discussions#88] [Ideas]
Since we have one set of dirs to rule them all, would be nice to be able to use the centralized location to validate and generate roms sets for imaging cards for romcarts (until wireless options become available). Basically, https://github.com/frederic-mahe/Hardware-Target-Game-Database integration such that one set of base roms can be used to construct whatever sets are appropriate for whatever carts.

**Comment by danmons** [comment]
Spoiler! :) Been working on this silently all week. 

Hope to have a public commit this weekend. Will need lots of community testing.

**Comment by danmons** [comment]
Alpha quality, lots of missing systems still, but the skeleton is committed now.  Please test and give feedback:
https://github.com/danmons/retronas/wiki/ROM-import

**Comment by citizen-why** [comment]
While still at Alpha Quality a useful feature to support testing would be a feature select on which SMBD packs to analyze against on that particular run. 

Use Case - I'm trying to test the project against the SegaCD pack - but since each pack does a full analyze of all files (900+), and SegaCD is near then end of the processing list - its taking quite sometime to be able to test results.

Not mission critical but could be a nice ease of use enhancement especially while testing and as new SMBD packs are integrated

**Reply by retinoic** [reply]
Another useful feature to speed up the process when parsing the import dir for multiple system SMDBs would be to move already-processed/successfully copied files out of the import directory (or into an ignored sub-directory), after each SMDB analysis so they aren't scanned repeatedly for each system after already being matched.

**Reply by sairuk** [reply]
I have revisited the romimport tool and you can now target a system with `-t`. You can list available target ids with `-l`. You'll need to run this yourself from the terminal, there is no gui for it.

```
# /opt/retronas/scripts/romimport.sh -l
Use id with -t to run a single system
-------------------------------------
id: atari/2600                  Atari 2600 SMDB.txt
id: nec/pcenginecd              PC Engine CD Redump Supplement.txt
id: nintendo/nintendo64         EverDrive 64 & 64drive SMDB.txt
id: nintendo/nes                EverDrive N8 & PowerPak SMDB.txt
id: sega/saturn                 Saturn Redump Supplement.txt
id: sega/megadrive              Mega EverDrive SMDB.txt
id: sega/megacd                 SegaCD.txt
id: sega/mastersystem           Master EverDrive SMDB.txt
id: atari/lynx                  Atari Lynx SMDB.txt
id: nintendo/gameboyadvance     EverDrive GBA SMDB.txt
id: nintendo/gameboycolor       Game Boy Color SMDB.txt
id: atari/7800                  Atari 7800 SMDB.txt
id: nintendo/gameboy            Game Boy SMDB.txt
id: sega/gamegear               EverDrive GG SMDB.txt
id: nintendo/snes               Super EverDrive & SD2SNES SMDB.txt
id: atari/5200                  Atari 5200 SMDB.txt
id: nec/pcengine                Turbo EverDrive & Super SD System 3 SMDB.txt
id: other/coleco_colecovision   Colecovision SMDB.txt
id: nintendo/famicom_disk       Famicom Disk System SMDB.txt
```

```
# /opt/retronas/scripts/romimport.sh -t nec/pcengine
------------------------
Analysing with ./EverDrive Pack SMDBs/Turbo EverDrive & Super SD System 3 SMDB.txt
Outputting to /data/retronas/roms/nec/pcengine
processing file:         0 / 0
coverage: 0/1467 (0.0%)
------------------------
```

In regards to moving already processed items from the directory that will need to be raised against smoke's tooling, we offload the actual directory content processing to the existing tool, as fair as I can see there is no mode to support moving directory contents after processing. We are currently using the smart file-strategy

**Reply by retinoic** [reply]
nice update! I ended up running the tool manually & targeting specific sub-directories.  Maybe allowing a specific directory to be passed to the script could be useful....

i.e. ./romimport.sh -t atari/2600 -f /storage/romimport/2600
